templates/rabbitmq.config.erb

```diff
@@ -153,7 +153,11 @@
 % Configure the LDAP authentication plugin
   {rabbitmq_auth_backend_ldap, [
     {other_bind, <%= @ldap_other_bind %>},
+<% if @ldap_server.class == Array or @ldap_server =~ /^\[/  -%>
```

Comment:
I usually see `@foo.is_a?(Array)` instead?
As mentioned in the comments, my personal take is that it's probably better not to support both an array and a string formatted as an array, and then simplify this to a single bit in the template, like
`{servers, [<%= @ldap_server.map { |v| "#{v}" }.join(', ') %>]},`

Reply:
That sounds okay to me, however I will need to retain the "flattened string" detection (and therefore maintain my own fork) for now as our use case is based around the fact that hiera squashes all data types to a string (https://tickets.puppetlabs.com/browse/PUP-7675); though I do appreciate that you won't necessarily want that in a released version. Any advice on how to get around that problem would be greatly appreciated though...

Reply:
@skrussell I've never had problems defining arrays in hiera. Should be just like:
```
rabbitmq::ldap_server:
  - server1.example.com
  - server2.example.com
```

Reply:
Yup, and that's how I've tested it, however, if you have a module that dynamically defines a variable somewhere, like we have, and then you use the heira scope variable lookup, then the array is flattened to a string.

So, if you have this manifest (this is a simplified example, but assume the servers variable is dynamically populated for now):
{code}
class ldap {
  $servers = [ 'svr1.example.com', 'svr2.example.com' ]
}
{code}
and in hiera:
{code}
rabbitmq::ldap_server: %{ldap::servers}
{code}
The value passed to the rabbitmq parameter gets squashed to a string rather than being passed as an array. This is a limitation of the scope() function that is being used by hiera, and will hopefully be fixed in a future release, but until this, this is a pain for us as we want to be able to dynamically allocate (the nearest) ldap servers without statically defining lists per node in Hiera.

Reply:
I use `[@ldap_servers].flatten` quite a lot to support both string and array[string]

Reply:
I refactored our code to remove the need to support an array as string, so now we just use a straight forward array (like was originally suggested/requested).

If we could look again at merging this for the next release, that would be great.